### PR TITLE
Proposal fix for  #166 

### PR DESCRIPTION
**Pull request by lablnet**
Proposal fix for  #166 
The issue with this proposal the Dispatch method become duplicate in
Zest Router class => https://github.com/zestframework/Zest_Framework/blob/master/src/Router/Router.php#L412
and in Component class {this pull request #167}

### CODE REVIEW

**Comment by lablnet** [comment]
@peter279k  What do you think?

File: src/Component/Component.php [hunk 1]
```diff
@@ -16,6 +16,53 @@
...
+                    $action = $this->params['action'];
+                    $action = $this->convertToCamelCase($action);
+                    if (preg_match('/action$/i', $action) == 0) {
+                        $controller_object->$action();
+                        (isset($this->params['middleware']) && is_object($this->params['middleware'])) ? (new $this->params['middleware']())->after(new Request(), new Response(), $this->params) : null;
+                    } else {
+                        throw new \Exception("Method $action in controller $controller cannot be called directly - remove the Action suffix to call this method");
+                    }
+                } else {
+                    throw new \Exception("Controller class $controller not found");
```
Comment (peter279k) [inline src/Component/Component.php:58]: If the route is not matched, it should send the `404` HTTP status code.

It means the request URL is not found.

Comment (peter279k) [reply src/Component/Component.php:58]: I don't think that throwing exception directly is a good way to solve this issue.

Comment (lablnet) [reply src/Component/Component.php:58]: yes, but what about if user want to customize this like in Laravel
http://laravel.com/dadasd

Comment (lablnet) [reply src/Component/Component.php:58]: oh @peter279k sorry for misunderstand we can fetch Views from View folder

Comment (peter279k) [reply src/Component/Component.php:58]: Just return view with customized error page :).

Comment (peter279k) [reply src/Component/Component.php:58]: Perhaps we can pass the default error page with view.

And let users customize their error pages.

Comment (lablnet) [reply src/Component/Component.php:58]: @peter279k  thanks

**Pull request merged**